### PR TITLE
Added TemplateBasedPresetExclusion

### PR DESCRIPTION
**Pull request by alan-null**
This pull request brings support for template id exlusions

Example configuration:
```xml
<include database="master" path="/sitecore/media library/Default Website">
	<exclude templateId="{3B4F2B85-778D-44F3-9B2D-BEFF1F3575E6}" />
</include>
```
Originally implemented [here](https://alan-null.github.io/2017/03/custom-unicorn-predicate)

### CODE REVIEW

File: src/Unicorn.Tests/Predicates/TestConfiguration.xml [hunk 1]
```diff
@@ -25,6 +25,17 @@
 		<exclude path="/implicit-nochildren/" /> <!-- implicit exclusion of a child path. NOTE: does not work for paths under the root (e.g. /implicit-nochildren/foo/) -->
 	</include>
 
+  <!-- 
+		TEMPLATE-ID: Exclude all children with a given templateId
```
Comment (kamsar) [inline src/Unicorn.Tests/Predicates/TestConfiguration.xml]: Add a note that children of items with the excluded template ID will also be excluded

File: src/Unicorn.Tests/Predicates/SerializationPresetPredicateTests.cs [hunk 2]
```diff
@@ -37,8 +38,13 @@ public void ctor_ThrowsArgumentNullException_WhenNodeIsNull()
 		[InlineData("/implicit-nochildren/ignoredchild", false)]
 		[InlineData("/implicit-nochildren/ignored/stillignored", false)]
 		[InlineData("/implicit-nochildrenwithextrachars", false)]
-		// SOME-CHILDREN test config
-		[InlineData("/somechildren", true)]
+        // TEMPLATE-ID test config
```
Comment (kamsar) [inline src/Unicorn.Tests/Predicates/SerializationPresetPredicateTests.cs]: Check your whitespace; it looks like tabs have converted to spaces in most of the code.

There is an editorconfig file that defines the project style; VS2015 has a plugin that can follow this; VS2017 follows it natively.

File: src/Unicorn.Tests/Predicates/TestConfiguration.xml (hunk 1 above)
Comment (alan-null) [inline src/Unicorn.Tests/Predicates/TestConfiguration.xml]: Additional note has been added.

File: src/Unicorn.Tests/Predicates/SerializationPresetPredicateTests.cs (hunk 2 above)
Comment (alan-null) [inline src/Unicorn.Tests/Predicates/SerializationPresetPredicateTests.cs]: yeah, I was using my default settings with `show-whitespaces` disabled in SourceTree. Now everything should be fine.

**Comment by alan-null** [comment]
@kamsar 
Thanks. Done.

**Comment by kamsar** [comment]
Thank you for your contribution :)

**Pull request merged**